### PR TITLE
Track chat lifecycle events

### PR DESCRIPTION
**Pull request by brunadafonseca**

### CODE REVIEW

**Review by phortuin — COMMENTED** [review]
Is it possible to make the `track` method a separate component? So refactor `opt-in.vue` and `lazy-chat.vue` to have access to a plugin or module that handles ga events?

File: src/client/components/opt-in/opt-in.vue [hunk 1]
```diff
@@ -52,9 +52,18 @@ export default {
         localStorage.setItem(this.name, true)
       }
       this.$emit('accept')
+      this.track('accepted')
     },
     onDecline () {
       this.$emit('decline')
+      this.track('declined')
+    },
+    track (label) {
```
Comment (jbmoelker) [inline src/client/components/opt-in/opt-in.vue]: `label` -> `eventLabel`?

File: src/client/components/lazy-chat/lazy-chat.vue [hunk 2]
```diff
@@ -9,7 +9,7 @@
     <button v-if="chatButtonIsShown"
```
Comment (jbmoelker) [inline src/client/components/lazy-chat/lazy-chat.vue:9]: `type="button"`

File: src/client/components/lazy-chat/providers/crisp.js [hunk 3]
```diff
@@ -18,6 +20,14 @@ export function onAccepted () {
  */
 export function onLoaded () {}
 
+function onOpened () {
```
Comment (jbmoelker) [inline src/client/components/lazy-chat/providers/crisp.js]: I'd rather keep the `ga` implementation out of the `crisp` provider. So can do to the same here as you do `onSessionLoaded` in the chat icon feature? Expose `onOpened` and `onClosed` as methods on this provider and make them accept a callback. In `lazy-chat` you can then use these methods and do the tracking.

File: src/client/components/lazy-chat/lazy-chat.vue [hunk 4]
```diff
@@ -47,6 +47,16 @@ export default {
     onLoaded () {
       this.isLoaded = true
       this.provider.onLoaded()
+    },
+    handleClick () {
+      this.isRequested = true
+      this.track()
+    },
+    track () {
```
Comment (jbmoelker) [inline src/client/components/lazy-chat/lazy-chat.vue]: To be consistent I'd set the `eventAction` to `click` and the `eventLabel` to `start`?
And actually, I'd change the signature of this method to `track(eventLabel)` like you do in `opt-in`. That way you can use it in `handleClick` and also `onOpened` and `onClosed` as suggested in my comment in `providers/crisp.js`

**Review by jbmoelker — CHANGES_REQUESTED** [review]

**Review by jbmoelker — APPROVED** [review]

**Pull request merged**